src/PuliPluginImpl.php

```diff
@@ -116,38 +113,19 @@ public function __construct(Event $event, PuliRunner $puliRunner = null)
         $this->isDev = $event->isDevMode();
         $this->puliRunner = $puliRunner;
         $this->rootDir = Path::normalize(getcwd());
-
-        $vendorDir = $this->config->get('vendor-dir');
-
-        // On TravisCI, $vendorDir is a relative path. Probably an old Composer
-        // build or something. Usually, $vendorDir should be absolute already.
-        $vendorDir = Path::makeAbsolute($vendorDir, $this->rootDir);
-
-        $this->autoloadFile = $vendorDir.'/autoload.php';
     }
 
     public function preAutoloadDump()
     {
-        if (!$this->initialized) {
-            $this->initialize();
-        }
-
-        // This method is called twice. Run it only once.
+        // This method is called multiple times. Run it only once.
         if (!$this->runPreAutoloadDump) {
             return;
         }
 
         $this->runPreAutoloadDump = false;
 
-        try {
-            $factoryClass = $this->getConfigKey('factory.in.class');
-            $factoryFile = $this->getConfigKey('factory.in.file');
-        } catch (PuliRunnerException $e) {
-            $this->printWarning('Could not load Puli configuration', $e);
-
-            return;
-        }
-
```

Comment:
Hope the new version makes that more clear. The code executed does not throw a PuliRunnerException any more, thus the try/catch does not make sense any more. The exceptions that might be triggered really are fatals (mainly programming mistakes in the plugin), which should bubble up.